### PR TITLE
Callbacks & Loggers

### PR DESCRIPTION
**Pull request by kesne**
So this PR covers a few things:

1.) Included new logger class. Overwriting the logger can be done by passing a logger param into the config object.
2.) Moved Eureka to a named export.
3.) Added callbacks to all async functions. Callbacks follow node style. If no callback is passed it defaults to a noop.
4.) No longer calling register and fetchRegistry out of the constructor.
5.) New start method that will run the register and the fetchRegistry in parallel and fire the callback when it is complete.

### CODE REVIEW

**Comment by kesne** [comment]
Also yes I know the coverage decreased, but this makes things more testable, so in reality it will increase once this gets in.

**Comment by kesne** [comment]
Fixes #6, #8, #11.

File: src/Logger.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    if (this._level >= LEVELS[method]) {
+      console[method](...args);
+    }
+  }
+
+  error() {
+    return this._log('error', arguments);
+  }
+  warn() {
+    return this._log('ward', arguments);
```
Comment (jquatier) [inline src/Logger.js]: ward?

Comment (kesne) [inline src/Logger.js]: Sometimes my love for coworkers bleeds into my work. Fixing.

**Comment by jquatier** [comment]
@kesne update the readme.

**Comment by kesne** [comment]
Will do. I'll go ahead and add some more documentation while I'm at it.

**Pull request merged**